paimon-core/src/main/java/org/apache/paimon/memory/MemoryPoolFactory.java

```diff
@@ -28,15 +32,26 @@ public class MemoryPoolFactory {
 
     private final MemorySegmentPool innerPool;
     private final int totalPages;
-    private final Iterable<MemoryOwner> owners;
 
-    public MemoryPoolFactory(MemorySegmentPool innerPool, Iterable<MemoryOwner> owners) {
+    private Iterable<MemoryOwner> owners;
+
+    public MemoryPoolFactory(MemorySegmentPool innerPool) {
         this.innerPool = innerPool;
         this.totalPages = innerPool.freePages();
-        this.owners = owners;
+    }
+
+    public MemoryPoolFactory addOwners(Iterable<MemoryOwner> newOwners) {
+        if (this.owners == null) {
+            this.owners = newOwners;
```

Comment:
add test here.

here should store `List<Iterable<MemoryOwner>>`.
If there is a new write generated later, it will not enter this factory